tests/core/CoreTestBase.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from coalib.core.Core import run
+
+
+class CoreTestBase(unittest.TestCase):
+    def execute_run(self, bears, executor=None):
+        """
+        Executes a coala run and returns the results.
+
+        This function has multiple ways to provide a different executor than
+        the default one (topmost item has highest precedence):
+
+        - Pass it via the ``executor`` parameter.
+        - Assign an executor class and the according instantiation arguments to
+          ``self.executor`` during ``setUp()``.
+
+        :param bears:
+            The bears to run.
+        :param executor:
+            The executor to run bears on.
+        :return:
+            A list of results.
+        """
+        if executor is None:
+            if hasattr(self, 'executor'):
```

Comment:
`executor = getattr(self, 'executor', None)` will avoid the `else` branch, and will allow callers to use `self.executor = None`

Reply:
I did now this:
```python
        if executor is None:
            executor = getattr(self, 'executor', None)
            if executor is not None:
                cls, args, kwargs = self.executor
                executor = cls(*args, **kwargs)
```
I hope that's fine :)